test/functional/test-intersection-observer-polyfill.js

```diff
@@ -0,0 +1,397 @@
+          intersectionRect: DomRectLtwh(50, 0, 50, 100),
+          intersectionRatio: 1 / 6,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('intersects on the edge', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 200, 150, 200);
+        };
+        const rootBounds = layoutRectLtwh(0, 100, 100, 100);
+        io.observe(element);
+        io.prevThresholdSlot_ = -1;
+        io.tick(rootBounds);
+        const change = {
+          time: 100,
+          rootBounds: DomRectLtwh(0, 0, 100, 100),
+          boundingClientRect: DomRectLtwh(50, 100, 150, 200),
+          intersectionRect: DomRectLtwh(50, 100, 50, 0),
+          intersectionRatio: 0,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('intersect correctly 2', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 199, 150, 200);
+        };
+        const rootBounds = layoutRectLtwh(0, 100, 100, 100);
+        io.observe(element);
+        io.tick(rootBounds);
+        const change = {
+          time: 100,
+          rootBounds: DomRectLtwh(0, 0, 100, 100),
+          boundingClientRect: DomRectLtwh(50, 99, 150, 200),
+          intersectionRect: DomRectLtwh(50, 99, 50, 1),
+          intersectionRatio: 1 / 600,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('intersect correctly 3', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 100, 150, 200);
+        };
+        const rootBounds = layoutRectLtwh(198, 299, 100, 100);
+        io.observe(element);
+        io.tick(rootBounds);
+        const change = {
+          time: 100,
+          rootBounds: DomRectLtwh(0, 0, 100, 100),
+          boundingClientRect: DomRectLtwh(-148, -199, 150, 200),
+          intersectionRect: DomRectLtwh(0, 0, 2, 1),
+          intersectionRatio: 2 / 30000,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('intersect correctly 4', () => {
+        element.getLayoutBox = () => {
+          return layoutRectLtwh(50, 100, 150, 200);
+        };
+        const rootBounds = layoutRectLtwh(202, 299, 100, 100);
+        io.observe(element);
+        io.prevThresholdSlot_ = -1;
+        io.tick(rootBounds);
+        const change = {
+          time: 100,
+          rootBounds: DomRectLtwh(0, 0, 100, 100),
+          boundingClientRect: DomRectLtwh(-152, -199, 150, 200),
+          intersectionRect: DomRectLtwh(0, 0, 0, 0),
+          intersectionRatio: 0,
+        };
+        expect(callbackSpy).to.be.calledOnce;
+        expect(callbackSpy).to.be.calledWith(change);
+      });
+
+      it('does not intersect with an element out of viewport', () => {
```

Comment:
please also add test for an InOb with threshold [0, 1], to make sure 
- 1st tick with 0 doesn't fire, 
- 2nd tick with 0.1 does fire
- 3rd tick with 0.9 doesn't fire
- 4rd tick with 1 does fire 
- 5th tick with 1 doesn't fire
- 6th tick with 0.9 does fire
- 7th tick with 0.1 doesn't fire
- 8th tick with 0 does fire
- 9th tick with 0 doesn't fire